airbyte-integrations/connectors/destination-bigquery/src/main/kotlin/io/airbyte/integrations/destination/bigquery/write/standard_insert/BigqueryBatchStandardInsertLoader.kt

```diff
@@ -162,14 +173,54 @@ class BigqueryBatchStandardInsertsLoaderFactory(
                 .setProject(bigquery.options.projectId)
                 .build()
 
+        val formatter: RecordFormatter =
+            when (dataChannelFormat) {
+                DataChannelFormat.PROTOBUF -> {
+                    if (config.legacyRawTablesOnly) {
+                        throw ConfigErrorException("Raw tables not supported in speed mode")
+                    }
+                    ProtoToBigQueryStandardInsertRecordFormatter(
+                        catalog.getStream(streamDescriptor).airbyteValueProxyFieldAccessors,
+                        tableNameInfo.columnNameMapping,
+                        catalog.getStream(streamDescriptor)
+                    )
+                }
+                else -> {
+                    BigQueryRecordFormatter(
+                        tableNameInfo.columnNameMapping,
+                        legacyRawTablesOnly = config.legacyRawTablesOnly,
+                    )
+                }
+            }
+
         return BigqueryBatchStandardInsertsLoader(
             bigquery,
             writeChannelConfiguration,
             jobId,
-            BigQueryRecordFormatter(
-                tableNameInfo.columnNameMapping,
-                legacyRawTablesOnly = config.legacyRawTablesOnly,
-            ),
+            formatter,
+        )
+    }
+
+    private fun <S> waitForStateStore(
+        stateStore: StreamStateStore<S>,
+        streamDescriptor: DestinationStream.Descriptor
+    ): S {
+        // Poll the state store until it's populated by the coordinating StreamLoader thread
+        var attempts = 0
+        val maxAttempts = 300
```

Comment:
when does this timer start running? It seems possible that stream setup could take longer than 5 minutes (i.e. if we need to execute a large schema change)

Reply:
do we need a timeout at all? I feel like if we're hanging in weird ways, that's just a bug in the CDK, and we shouldn't rely on connectors to be defensive about it?

Reply:
Ummmm! Lets still put a timeout otherwise it might be difficult to figure out that there is an issue! Will increase it to an hour!